src/Modules/Visualization/ShowColorMapModule.cc

```diff
@@ -193,119 +201,236 @@ ShowColorMapModule::buildGeometryObject(ColorMapHandle cm, ModuleStateHandle sta
-      new CPM_VAR_BUFFER_NS::VarBuffer(vboSize));
-    std::shared_ptr<CPM_VAR_BUFFER_NS::VarBuffer> vboBufferSPtr2(
-      new CPM_VAR_BUFFER_NS::VarBuffer(iboSize));
-
-    CPM_VAR_BUFFER_NS::VarBuffer* iboBuffer2 = iboBufferSPtr2.get();
-    CPM_VAR_BUFFER_NS::VarBuffer* vboBuffer2 = vboBufferSPtr2.get();
-
-    for (auto a : indices) iboBuffer2->write(a);
-    for (size_t i = 0; i < points.size(); i++) {
-      vboBuffer2->write(static_cast<float>(points[i].x()));
-      vboBuffer2->write(static_cast<float>(points[i].y()));
-      vboBuffer2->write(static_cast<float>(points[i].z()));
-      vboBuffer2->write(static_cast<float>(txt_coords[i].x()));
-      vboBuffer2->write(static_cast<float>(txt_coords[i].y()));
-    }
 
-    //add the actual points and colors
-
-    uniqueNodeID = id + "colorMapLegendTextFont" + ss.str();
-    vboName = uniqueNodeID + "VBO";
-    iboName = uniqueNodeID + "IBO";
-    passName = uniqueNodeID + "Pass2";
-
-    // NOTE: Attributes will depend on the color scheme. We will want to
-    // normalize the colors if the color scheme is COLOR_IN_SITU.
-
-    // Construct VBO.
-    shader = "Shaders/ColorMapLegendText";
-    attribs.clear();
-    attribs.push_back(SpireVBO::AttributeData("aPos", 3 * sizeof(float)));
-    attribs.push_back(SpireVBO::AttributeData("aTexCoord", 2 * sizeof(float)));
-    uniforms.clear();
-    uniforms.push_back(SpireSubPass::Uniform("uXTranslate", static_cast<float>(xTrans)));
-    uniforms.push_back(SpireSubPass::Uniform("uYTranslate", static_cast<float>(yTrans)));
-    uniforms.push_back(SpireSubPass::Uniform("uDisplaySide", static_cast<float>(displaySide)));
-    uniforms.push_back(SpireSubPass::Uniform("uDisplayLength", static_cast<float>(displayLength)));
-    uniforms.push_back(SpireSubPass::Uniform("uColor", glm::vec4(red, green, blue, 1.0f)));
-    SpireVBO geomVBO2 = SpireVBO(vboName, attribs, vboBufferSPtr2,
-      numVBOElements, BBox(), true);
-
-    geom->mVBOs.push_back(geomVBO2);
-
-    // Construct IBO.
-
-    SpireIBO geomIBO2(iboName, SpireIBO::TRIANGLES, sizeof(uint32_t), iboBufferSPtr2);
-    geom->mIBOs.push_back(geomIBO2);
-    renState.set(RenderState::USE_COLORMAP, false);
-    renState.set(RenderState::USE_TRANSPARENCY, true);
-    SpireSubPass pass2(passName, vboName, iboName, shader,
-      COLOR_MAP, renState, RENDER_VBO_IBO, geomVBO2, geomIBO2);
-
-    // Add all uniforms generated above to the pass.
-    for (const auto& uniform : uniforms) { pass2.addUniform(uniform); }
-    //******************************************************************************************
-    // TODO we're not adding this geometry (font) until we debug for it to work on Windows.
-    geom->mPasses.push_back(pass2);
-    //******************************************************************************************
+      //add the actual points and colors
+      uniqueNodeID = id + "colorMapLegendTextFont" +
+        static_cast<std::ostringstream*>(&(std::ostringstream()
```

Comment:
@basisunus Scratch that, I see what they're trying to do, and it's a really horrible way to construct a string. Please change to use a declared ostringstream object, better yet, write a function to stringify the values.